### PR TITLE
Fix the new workflow and Optimise it

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Issue #15567

### CODE REVIEW

**Comment by romani** [comment]
Please show prove of testing.

File: .github/workflows/regression-report.yml [hunk 1]
```diff
@@ -557,38 +556,60 @@ jobs:
...
     runs-on: ubuntu-latest
-    needs: [ check_pr_status, parse_comment, download_configs, make_report ]
-    if: failure() || success()
+    needs: [
+      handle_generated_config_bundle,
+      download_configs,
+      make_report ]
+    if: always()
+      && (needs.make_report.outputs.message != ''
+      || needs.download_configs.result == 'failure'
```
Comment (romani) [inline .github/workflows/regression-report.yml:565]: Put `||` under "n" or previous line

**Review by romani — CHANGES_REQUESTED** [review]
Minor

File: .github/workflows/regression-report.yml (hunk 1 above)
Comment (relentless-pursuit) [reply .github/workflows/regression-report.yml]: done

**Comment by relentless-pursuit** [comment]
> #15567

https://github.com/checkstyle-gsoc/checkstyle/pull/23#issuecomment-2315801342

Please refer the PR form here, can be bit messy if you analyse form the top. 

I have tested success scenarios, and failure scenarios.

**Review by romani — APPROVED** [review]
Ok to merge

File: .github/workflows/regression-report.yml [hunk 2]
```diff
@@ -557,38 +556,60 @@ jobs:
...
           JOBS_LINK="https://github.com/checkstyle/checkstyle/actions/runs/${{github.run_id}}"
-          if [ "$DOWNLOAD_FAILED" == "true" ]; then
-            echo "Report generation failed. Please use 'GitHub, generate report for" \
-                 "check_name/example_type'<br>Link: $JOBS_LINK" > .ci-temp/message
+          if [ "$DOWNLOAD_RESULT" == "failure" ]; then
+            {
+              echo "Failed to download or process the specified configuration(s)."
+              echo "Error details: $DOWNLOAD_ERROR"
+              echo "<br>Please ensure you've used one of the following commands correctly:"
+              echo "- 'GitHub, generate report for <config_name>'"
```
Comment (romani) [inline .github/workflows/regression-report.yml:585]: `<config_bundke_path>`

**Review by romani — COMMENTED** [review]
Please fix in follow up PR

**Pull request merged**